src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -222,7 +222,7 @@ public override INamedTypeSymbol CreateErrorTypeSymbol(INamespaceOrTypeSymbol co
             return Create(
                 assemblyName,
                 options ?? s_defaultSubmissionOptions,
-                (syntaxTree != null) ? new[] { syntaxTree } : SpecializedCollections.EmptyEnumerable<SyntaxTree>(),
+                (syntaxTrees != null) ? syntaxTrees.Cast<SyntaxTree>() : null,
```

Comment:
Copy paste mistake...removed here and above...

The cast should only happen in ```AddSyntaxTrees```.